### PR TITLE
FIX: Return correct shape of probabilities from GPU logreg

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR fixes the shape of the predictions produced by GPU logistic regression to match what the CPU version would produce.

Note that method `decision_function` is explicitly not tested as it's currently not implemented for GPU - will be left for a later PR.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

File: sklearnex/linear_model/tests/test_logreg.py [hunk 1]
```diff
@@ -196,3 +196,26 @@ def logistic_model_function(predicted_probabilities, coefs):
...
+
+
+@pytest.mark.parametrize(
+    "dataframe,queue", get_dataframes_and_queues(device_filter_="gpu")
+)
+def test_gpu_logreg_prediction_shapes(dataframe, queue):
+    from sklearnex.linear_model import LogisticRegression
+
+    X, y = make_classification(random_state=123)
+    X = X.astype(np.float32)
```
Comment (icfaust) [inline sklearnex/linear_model/tests/test_logreg.py]: How come float32?

Comment (david-cortes-intel) [reply sklearnex/linear_model/tests/test_logreg.py]: This is only for GPU.

Comment (icfaust) [reply sklearnex/linear_model/tests/test_logreg.py]: you can check the queue for has_aspect_64, cards like the Intel MAX 1550 support float64.

Comment (david-cortes-intel) [reply sklearnex/linear_model/tests/test_logreg.py]: Modified to use the same kind of filtering logic as used in other tests.
